scipy/ndimage/_filters.py

```diff
@@ -68,12 +68,23 @@ def _vectorized_filter_iv(input, function, size, footprint, output, mode, cval,
     if size is not None and footprint is not None:
         raise ValueError("Either `size` or `footprint` may be provided, not both.")
 
-    # Either footprint or size must be provided, and these determine the core
-    # dimensionality...
+    if axes is None:
+        axes = tuple(range(-input.ndim, 0))
+    elif np.isscalar(axes):
+        axes = (axes,)
+    n_axes = len(axes)
+    n_batch = input.ndim - n_axes
```

Comment:
We now ensure that `axes` is always a tuple from the outset. If `axes` is `None`, the default is that all dimensions are core dimensions; otherwise `axes` determines the number of core/batch dimensions.